packages/terra-base/src/Base.jsx

```diff
@@ -1,38 +1,75 @@
 import React, { PropTypes } from 'react';
-import './Base.scss';
+
+import { I18nProvider, i18nLoader } from 'terra-i18n';
+
+import BaseStyles from './BaseStyles';
 
 const propTypes = {
   /**
    * The component(s) that will be wrapped by `<Base />`.
    */
   children: PropTypes.node.isRequired,
+  /**
+   * The locale name.
+   */
+  locale: PropTypes.string.isRequired,
+  /**
+   * Customized translations provided by consuming application
+   * only for current locale.
+   */
+  customMessages: PropTypes.object,
+};
+
+const defaultProps = {
+  customMessages: {},
 };
 
 class Base extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      load: false,
+      locale: props.locale,
+      messages: {},
+    };
+  }
 
   componentDidMount() {
-    // Checks to run when not in production
-    if (process.env.NODE_ENV !== 'production') {
-      // Check to ensure terra-Base class is set on html element
-      if (!new RegExp('(^|\\s)terra-Base(\\s|$)').test(document.documentElement.className)) {
-        // eslint-disable-next-line
-        console.warn('The html element is missing the terra-Base class.');
-      }
-
-      // Check to ensure dir attribute is set on html element
-      if (!document.documentElement.hasAttribute('dir')) {
-        // eslint-disable-next-line
-        console.warn('The html element is missing the dir attribute. For terra directionality based styles to render correctly, add dir="ltr" or dir="rtl" to the html element.');
-      }
-    }
+    i18nLoader(this.props.locale, this.setState, this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props === nextProps) return;
+    i18nLoader(nextProps.locale, this.setState, this);
   }
 
   render() {
-    const { children, ...customProps } = this.props;
-    return <div {...customProps}>{children}</div>;
+    if (!this.state.load) return null;
+
+    const {
+      children,
+      locale,
```

Comment:
Yeah. The props locale is used to initialize the `state.locale`. `state.locale` is used to guarantee all required locale date has loaded before the Base component can be rendered. So in [constructor](https://github.com/cerner/terra-core/pull/231/files/1a59225243ab3e8053b088359f6b5b397f6aa398#diff-32b703eb84aad852d11b77791c8ff8e0R32), we use props.locale and in the render, we use `state.locale`. 

To explain the difference between `props.locale` and `state.locale`, this code may be helpful [en locale data](https://github.com/cerner/terra-core/blob/master/packages/terra-i18n/src/translations/en.js#L12). The `state.locale` is coming from the translations data. That will update the `state.locale` as `state.messages`.